app/plugins/Files/js/buttons.js

```diff
@@ -127,6 +129,43 @@ eID('cancel-ascii').onclick = function() {
 	hide('show-ascii');
 };
 
+// Select directory sliding frame
+eID('new-dir').onclick = function() {
+	show('add-dir');
+	hide('file-library');
+};
+eID('back-dir').onclick = exitFileAdder;
+
+// Upload directory option chosen
+eID('upload-dir-choice').onclick = function() {
+	var path = require("path");
+	hideSteps([21,31]);
+
+	var loadPath = IPC.sendSync('dialog', 'open', {
+		title: 'Select Directory',
+		properties: ['openDirectory'],
+	});
+	if (loadPath) {
+		eID('nickname-dir').querySelector('.dir-path').innerHTML = loadPath;
+		loadPath = loadPath[0].split(path.sep);
+		eID('nickname-dir-input').value = loadPath[loadPath.length - 1] + '_';
+		show('nickname-dir');
+		show('upload-dir');
+		eID('nickname-dir-input').focus();
+	}
+};
```

Comment:
This should exhibit some feedback when the loadPath is wrong, right? That can definitely happen with the Windows file browser at least.